test/terraform/mzcompose.py

```diff
@@ -320,24 +322,55 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
             ],
             cwd=path,
         )
-        # These sleeps are a bit unclean, we should instead only consider
-        # environmentd as running when it is has started listening on its ports
-        time.sleep(30)
-        port_forward_process = subprocess.Popen(
+        # Can take a while for balancerd to come up
+        for i in range(120):
+            try:
+                balancerd_name = spawn.capture(
+                    [
+                        "kubectl",
+                        "get",
+                        "pods",
+                        "-l",
+                        "app=balancerd",
+                        "-n",
+                        "materialize-environment",
+                        "-o",
+                        "jsonpath={.items[*].metadata.name}",
+                    ],
+                    cwd=path,
+                )
+                break
+            except subprocess.CalledProcessError as e:
+                print(e)
+                time.sleep(1)
+        else:
+            raise ValueError("Never completed")
+
+        environmentd_port_forward_process = subprocess.Popen(
             [
                 "kubectl",
                 "port-forward",
                 f"pod/{environmentd_name}",
                 "-n",
                 "materialize-environment",
-                "6875:6875",
```

Comment:
Should we also be doing the same with 6876?

Reply:
Sure, but we are currently not using that port in the test (we should in the future though)